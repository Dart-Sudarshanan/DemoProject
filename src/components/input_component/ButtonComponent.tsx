import React from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";

interface ButtonProps {
  testID: string;
  title: string;
  onPress: Function;
}

export const ButtonComponent: React.FC<ButtonProps> = ({ title, onPress, testID }) => {
  return (
    <View style={styles.container}>
      <Pressable testID={testID} style={styles.button} onPress={() => onPress()}>
        <Text style={styles.buttonTxt}>{title}</Text>
      </Pressable>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    margin: 10,
  },
  button: {
    paddingHorizontal: 40,
    paddingVertical: 10,
    borderRadius: 4,
    backgroundColor: "#4c51e6",
  },
  buttonTxt: {
    fontSize: 18,
    fontWeight: "500",
    color: "#fff",
  },
});
