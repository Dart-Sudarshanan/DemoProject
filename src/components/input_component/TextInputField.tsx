import React from "react";

import { StyleSheet, Text, TextInput, View } from "react-native";

const TextInputField = (props: any) => {
  return (
    <View style={styles.container}>
      <Text style={styles.labels}>{props.label}</Text>
      <View>
        <TextInput
          style={styles.textInput}
          {...props}
          value={props.input.value}
          onChangeText={props.input.onChange}
          onBlur={props.input.onBlur}
          onFocus={props.input.onFocus}
        />
      </View>
    </View>
  );
};

export default TextInputField;

const styles = StyleSheet.create({
  container: {
    margin: 10,
  },
  labels: {
    fontSize: 18,
    fontWeight: "400",
  },
  textInput: {
    borderWidth: 2,
    borderRadius: 8,
    borderColor: "#a39d9d",
    padding: 10,
  },
});
