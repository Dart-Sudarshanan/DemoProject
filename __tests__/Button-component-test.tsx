import React from "react";

import renderer from "react-test-renderer";
import { ButtonComponent } from "../src/components/input_component/ButtonComponent";

it("Button Component", () => {
  const tree = renderer.create(<ButtonComponent title='Test label' onPress={() => {}} />);
  expect(tree).toMatchSnapshot();
});
