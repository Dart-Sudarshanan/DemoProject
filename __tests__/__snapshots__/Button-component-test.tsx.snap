// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#4c51e6",
        "borderRadius": 4,
        "paddingHorizontal": 40,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Test label
    </Text>
  </View>
</View>
`;
